library/src/main/java/com/novoda/downloadmanager/lib/BatchQuery.java

```diff
@@ -211,29 +220,45 @@ public Builder withExtraData(String extraData) {
         }
 
         public BatchQuery build() {
-            if (criteriaIdBuilder != null) {
-                builder.withInnerCriteria(criteriaIdBuilder.build());
-                if (criteriaStatusBuilder != null) {
+
+            List<Criteria.Builder> criteriaBuilders = setCriteriaBuilders();
+            setCriteriaListFrom(criteriaBuilders);
+
+            Criteria criteria = builder.build();
+            String selection = criteria.getSelection();
+            String sortOrder = criteria.getSort();
+            String[] selectionArguments = criteria.getSelectionArguments();
+            return new BatchQuery(selection, selectionArguments, sortOrder);
+        }
+
+        private void setCriteriaListFrom(List<Criteria.Builder> criteriaBuilders) {
+            for (Criteria.Builder criteriaBuilder : criteriaBuilders){
+                builder.withInnerCriteria(criteriaBuilder.build());
+                if (isNotLast(criteriaBuilder, criteriaBuilders)) {
                     builder.and();
                 }
             }
+        }
 
+        private boolean isNotLast(Criteria.Builder criteriaBuilder, List<Criteria.Builder> criteriaBuilders) {
+            return criteriaBuilders.indexOf(criteriaBuilder) != (criteriaBuilders.size() - 1);
+        }
+
+        private List<Criteria.Builder> setCriteriaBuilders() {
```

Comment:
Better. Too much `set*` and `build*` :smirk:
https://github.com/novoda/download-manager/commit/7f71c4f5c16578217f2a5382649e105575fa71cd